types/mithril/index.d.ts

```diff
@@ -246,7 +246,7 @@ declare namespace Mithril {
 	 * Components are a mechanism to encapsulate parts of a view to make code easier to organize and/or reuse.
 	 * Any Javascript object that has a view method is a Mithril component. Components can be consumed via the m() utility.
 	 */
-	type Comp<Attrs, State extends Lifecycle<Attrs, State>> = Component<Attrs, State> & State;
```

Comment:
`m` is a (variadic) hyperscript function (the rest of the Mithril API is attached as properties to this function.) Users don't implement hyperscript themselves. I'll take a look at how the indexer would affect `Lifecycle` tonight or this weekend.